src/xwayland/xwm/mod.rs

```diff
@@ -889,31 +881,31 @@ fn handle_event<D: XwmHandler>(state: &mut D, xwmid: XwmId, event: Event) -> Res
         }
         Event::MapNotify(n) => {
             slog::trace!(xwm.log, "X11 Window mapped: {}", n.window);
-            if let Some(surface) = xwm.windows.iter().find(|x| x.window_id() == n.window).cloned() {
+            if let Some(surface) = xwm
+                .windows
+                .iter()
+                .find(|x| x.window_id() == n.window || x.state.lock().unwrap().mapped_onto == Some(n.window))
```

Comment:
Couldn't this use the `mapped_window_id` you added in one of the commits here? Or is this because of an Inner type?